parser-typechecker/src/Unison/Codebase/Editor/Propagate.hs

```diff
@@ -56,10 +57,12 @@ data Edits v = Edits
   , typeEdits :: Map Reference TypeEdit
   , typeReplacements :: Map Reference Reference
   , newTypes :: Map Reference (Decl v Ann)
+  , constructorReplacements :: Map (Reference, Int, ConstructorType)
```

Comment:
The `Edits` type now tracks which constructors of old types are replaced with which constructors of new types. The mapping is based on the same thing as term and type replacements: on there being a common name between the two. The name we use is somewhat more stable though as it comes from the original `Decl`.